server/server.go

```diff
@@ -363,6 +366,31 @@ func (s *Server) SpeedTestHandler(w http.ResponseWriter, req *http.Request) erro
 	return nil
 }
 
+// SnapshotHandler forces etcd store to do a snapshot. If the disk parameter is set, the snapshot
+// will be written to disk at data-dir/index-migrate.snap. Or the snapshot will be returned as
+// http body.
+func (s *Server) SnapshotHandler(w http.ResponseWriter, req *http.Request) error {
+	data, err := s.Store().Save()
+	if err != nil {
+		http.Error(w, "failed to create snapshot:"+err.Error(), http.StatusInternalServerError)
+		log.Warn("Failed to create snapshot:" + err.Error())
+		return nil
+	}
+	if req.FormValue("disk") == "true" {
```

Comment:
Should the contents of this form value be checked for validity (and return a 400 otherwise)? Providing the value "True" , for example, will silently be ignored.